pkg/reconcile/pipeline/infinispan/context/context.go

```diff
@@ -58,12 +60,29 @@ type contextImpl struct {
 	*flowCtrl
 	*provider
 	*pipeline.ContextProviderConfig
-	ctx        context.Context
-	infinispan *ispnv1.Infinispan
-	ispnConfig *pipeline.ConfigFiles
-	ispnClient api.Infinispan
-	ispnPods   *corev1.PodList
-	resources  map[string]client.Object
+	ctx            context.Context
+	versionManager *version.Manager
+	infinispan     *ispnv1.Infinispan
+	ispnConfig     *pipeline.ConfigFiles
+	ispnClient     api.Infinispan
+	ispnPods       *corev1.PodList
+	resources      map[string]client.Object
+}
+
+func (c *contextImpl) Operand() version.Operand {
+	v := c.Infinispan.Spec.Version
+	if v == "" {
+		return c.versionManager.Latest()
+	}
+	if operand, err := c.VersionManager.WithRef(v); err != nil {
+		panic(fmt.Errorf("spec.Version '%s' doesn't exist", v))
```

Comment:
I used a `panic` here instead of returning `(version.Operand, error)` for the following reasons:

- It shouldn't be possible for `spec.Version` to be an invalid version due to the Webhook checks
- It simplifies the handler logic so we don't have to handle the returned `error` on every invocation of `Context#Operand` for a state that should never happen
- If a `panic` is thrown by a handler, then it's caught and logged by the pipeline, so the Operator will continue to function.